bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/AccountController.java

```diff
@@ -0,0 +1,163 @@
+        sendHttp(payload);
+    }
+
+    private void sendHttp(String payload) throws SunSynkAuthenticateException, SunSynkTokenException {
+        Gson gson = new Gson();
+        String response = "";
+        String httpsURL = makeLoginURL("oauth/token");
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Requester", "www.openhab.org"); // optional
+        InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+        try {
+            response = HttpUtil.executeUrl("POST", httpsURL, headers, stream, "application/json", 2000);
+            this.sunAccount = gson.fromJson(response, Client.class);
+        } catch (IOException | JsonSyntaxException e) {
+            throw new SunSynkAuthenticateException("Sun Synk Account could not be authenticated", e);
+        }
+        if (this.sunAccount == null) {
+            throw new SunSynkAuthenticateException("Sun Synk Account failed to get a response.");
+        }
+        if (this.sunAccount.getCode() == 102) {
+            logger.info("Sun Synk Account could not be authenticated: {}.", this.sunAccount.getMsg());
+            throw new SunSynkAuthenticateException(
+                    "Sun Synk Accountfailed to authenticate: Check your password or email.");
+        }
+        if (this.sunAccount.getStatus() == 404) {
+            logger.info("Sun Synk Account could not be authenticated: 404 {} {}.", this.sunAccount.getError(),
+                    this.sunAccount.getPath());
+            throw new SunSynkAuthenticateException("Sun Synk Accountfailed to authenticate: 404 Not Found.");
+        }
+        getToken();
+    }
+
+    private void getToken() throws SunSynkTokenException {
+        if (this.sunAccount.getData().getAccessToken().isEmpty()) {
+            throw new SunSynkTokenException("Sun Synk Account failed to get token");
+        }
+        APIdata.static_access_token = this.sunAccount.getData().getAccessToken();
+    }
+
+    public ArrayList<Inverter> getDetails() throws SunSynkInverterDiscoveryException {
+        @Nullable
+        Details output = new Details();
+        ArrayList<Inverter> inverters = new ArrayList<>();
+        try {
+            Gson gson = new Gson();
+            Properties headers = new Properties();
+            String response = "";
+            String httpsURL = makeLoginURL(
+                    "api/v1/inverters?page=1&limit=10&total=0&status=-1&sn=&plantId=&type=-2&softVer=&hmiVer=&agentCompanyId=-1&gsn=");
+            headers.setProperty("Accept", "application/json");
+            headers.setProperty("Authorization", "Bearer " + APIdata.static_access_token);
+            headers.setProperty("Requester", "www.openhab.org"); // optional
+            response = HttpUtil.executeUrl("GET", httpsURL, headers, null, "application/json", 2000);
```

Comment:
See other comment about `executeUrl` and http method + timeout

Reply:
Done